crates/ra_hir/src/expr.rs

```diff
@@ -792,7 +788,54 @@ impl ExprCollector {
             ast::ExprKind::Label(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::IndexExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::RangeExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
-            ast::ExprKind::MacroCall(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
+            ast::ExprKind::MacroCall(e) => {
+                // very hacky
+                let name = e
+                    .path()
+                    .map(|path| Path::from_ast(path))
+                    .unwrap()
```

Comment:
unwraps here could fire, the should be changed to matching. I think the best way to do this would be to extract an `expand_macro_expr` function which returns an `Option`, so that we can employ `?`